File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/domain/board/service/UpdateBoard.java
package com.example.mediumspringwebflux.domain.board.service;

import com.example.mediumspringwebflux.domain.board.document.Board;
import com.example.mediumspringwebflux.domain.board.dto.BoardRequest;
import com.example.mediumspringwebflux.domain.board.repository.BoardRepository;
import com.example.mediumspringwebflux.domain.board.service.facade.FindBoard;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
public class UpdateBoard {
    private final BoardRepository boardRepository;
    private final FindBoard findBoard;

    public UpdateBoard(BoardRepository boardRepository, FindBoard findBoard) {
        this.boardRepository = boardRepository;
        this.findBoard = findBoard;
    }

    public Mono<Void> execute(String id, BoardRequest request) {
        return Mono.defer(() -> findBoard.execute(id)
                .flatMap(board -> updateBoard(board, request)))
                .then();
    }

    private Mono<Board> updateBoard(Board board, BoardRequest request) {
        return Mono.fromRunnable(() -> {
            board.edit(request.title(), request.description());
        }).thenReturn(board);
    }

}

File: mediumSpringWebfluxcorutine/src/main/kotlin/com/example/mediumspringwebfluxcorutine/board/service/UpdateBoard.kt
package com.example.mediumspringwebfluxcorutine.board.service

import com.example.mediumspringwebfluxcorutine.board.document.Board
import com.example.mediumspringwebfluxcorutine.board.dto.BoardRequest
import com.example.mediumspringwebfluxcorutine.board.repository.BoardRepository
import com.example.mediumspringwebfluxcorutine.board.service.facade.FindBoard
import kotlinx.coroutines.reactor.awaitSingle
import org.springframework.stereotype.Service

@Service
class UpdateBoard(
    private val boardRepository: BoardRepository,
    private val findBoard: FindBoard
) {

    suspend fun execute(id: String,
                        boardRequest: BoardRequest) {
        val board: Board = findBoard.byId(id)

        board.edit(boardRequest)

        boardRepository.save(board)
            .awaitSingle()
    }

}